tests/pytests/test_vecsim.py

```diff
@@ -1016,3 +1020,48 @@ def test_hybrid_query_with_global_filters():
                'filter', 'num', 0, index_size/2, 'filter', 'num', index_size/3, index_size/2,
                'geofilter', 'coordinate', 5.0, 5.0, 50, 'km', 'SORTBY', '__v_score',
                'NOCONTENT', 'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+
+def test_hybrid_query_change_policy():
+    env = Env(moduleArgs = 'DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    dim = 2
+    n = 6000 * env.shardsCount
+    vectors = np.random.rand(n, dim).astype(np.float32)
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'v', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32',
+               'DIM', dim, 'DISTANCE_METRIC', 'COSINE', 'tag1', 'TAG', 'tag2', 'TAG').ok()
+
+    file = 1
+    tags = range(10)
+    subset_size = int(n/2)
+    with conn.pipeline(transaction = False) as p:
+        for i, v in enumerate(vectors[:subset_size]):
+            conn.execute_command('HSET', i, 'v', v.tobytes(),
+                                 'tag1', str(tags[randrange(10)]), 'tag2', 'word'+str(file))
+        p.execute()
+
+    file = 2
+    with conn.pipeline(transaction = False) as p:
+        for i, v in enumerate(vectors[subset_size:]):
+            conn.execute_command('HSET', i + subset_size, 'v', v.tobytes(),
+                                 'tag1', str(10 + tags[randrange(10)]), 'tag2', 'word'+str(file))
+        p.execute()
+
+    # This should return 10 results and run in HYBRID_BATCHES mode
```

Comment:
please add validation on the running mode

Reply:
This validation always occurs in `execute_hybrid_query`. I'll add the argument explicitly for more readability.